modules/gapi/test/own/matview_tests.cpp

```diff
@@ -0,0 +1,146 @@
+    expect_eq_desc(view, desc);
+    EXPECT_EQ(view.ptr(), nullptr);
+    EXPECT_EQ(view.step(), 0u);
+}
+
+struct MatViewTest : public TestWithParam<int>{};
+TEST_P(MatViewTest, ConstructionFromMat) {
+    auto type = GetParam();
+    Mat mat(8,8,type);
+    const auto desc = cv::descr_of(mat);
+    MatView view(cv::descr_of(mat), mat.ptr(), mat.step1());
+    expect_eq_desc(view, desc);
+    EXPECT_EQ(view.ptr(), mat.ptr());
+    EXPECT_EQ(view.step(), mat.step1());
+}
+
+TEST(MatView, TestConstructionFromMatND) {
+    std::vector<int> dims(4, 8);
+    Mat mat(dims, CV_8UC1);
+    const auto desc = cv::descr_of(mat);
+    MatView view(cv::descr_of(mat), mat.ptr());
+    expect_eq_desc(view, desc);
+    EXPECT_EQ(view.ptr(), mat.ptr());
+}
+
+TEST_P(MatViewTest, DefaultStep) {
+    auto type = GetParam();
+    GMatDesc desc;
+    desc.chan = CV_MAT_CN(type);
+    desc.depth = CV_MAT_DEPTH(type);
+    desc.size = {8,8};
+    std::vector<unsigned char> data(desc.size.width*desc.size.height*CV_ELEM_SIZE(type));
+    MatView view(desc, data.data());
+    EXPECT_EQ(view.step(), static_cast<size_t>(desc.size.width)*CV_ELEM_SIZE(type));
+}
+
+static Mat wrapViewByMat(MatView& view) {
+    return Mat(view.size(), view.type(), view.ptr(), view.step());
+}
+
+TEST_P(MatViewTest, StepIn) {
+    auto type = GetParam();
+    Mat bigMat(16,16,type);
+    cv::randn(bigMat, cv::Scalar::all(127), cv::Scalar::all(40));
+    Mat mat = bigMat(cv::Rect{4,4,8,8});
+    MatView view(cv::descr_of(mat), mat.data, mat.step);
+    const auto viewMat = wrapViewByMat(view);
+    Mat ref, out;
+    cv::Size ksize{1,1};
+    cv::blur(viewMat, out, ksize);
+    cv::blur(    mat, ref, ksize);
+    EXPECT_EQ(0, cvtest::norm(ref, out, NORM_INF));
+}
+
+TEST_P(MatViewTest, StepOut) {
+    auto type = GetParam();
+    Mat mat(8,8,type);
+    cv::randn(mat, cv::Scalar::all(127), cv::Scalar::all(40));
+    Mat bigMat = Mat::zeros(16,16,type);
+    Mat out = bigMat(cv::Rect{4,4,8,8});
+    MatView view(cv::descr_of(out), out.ptr(), out.step);
+    auto viewMat = wrapViewByMat(view);
+    Mat ref;
+    cv::Size ksize{1,1};
+    cv::blur(mat, viewMat, ksize);
+    cv::blur(mat, ref,     ksize);
+    EXPECT_EQ(0, cvtest::norm(ref, out, NORM_INF));
+}
+
+INSTANTIATE_TEST_CASE_P(Test, MatViewTest,
+                        Values(CV_8UC1, CV_8UC3, CV_32FC1));
+
+struct MatViewCallbackTest : public ::testing::Test {
+    MatViewCallbackTest()
+        : mat(8,8,CV_8UC1), view(cv::descr_of(mat), mat.ptr(), mat.step1(), [this](){ callbackCalls++; }) {
+        cv::randn(mat, cv::Scalar::all(127), cv::Scalar::all(40));
+    }
+    int callbackCalls = 0;
+    Mat mat;
+    MatView view;
+};
+
+TEST_F(MatViewCallbackTest, MoveCopy) {
+    {
+        MatView copy(std::move(view));
+        cv::util::suppress_unused_warning(copy);
+        EXPECT_EQ(callbackCalls, 0);
+    }
+    EXPECT_EQ(callbackCalls, 1);
+}
+
+static int firstElement(const MatView& view) { return *view.ptr(); }
```

Comment:
Why not using lambdas?

Reply:
> Why not using lambdas?

It was assumed that there can be more tests which use this functions